### DISCUSSION TITLE
delegating some URLs to another application

### DISCUSSION THREAD

**Discussion by keegancsmith** [qutebrowser/qutebrowser#8030] [Q&A]
I have a use case were some URLs I can only open in safari/chrome due to a required extension in those[^1]. Is it possible to detect those URLs and automatically handle them in another URL. It doesn't need to be whenever I browse to them, but rather only when I open them from another application.

I already have configured a quick shortcut to do this[^2], but the flow is a bit janky since I get redirected to my login flow instead of the respective URLs. This often ends up not working well.

I have considered writing a little wrapper script and making that the default browser. But that feels less ideal due to needing to make it behave like an app, so hoping there is a way in qutebrowser via its lovely config.py.

[^1]: Okta Verify hooks into the browser as an extra step to login to my work auth provider.
[^2]: https://github.com/keegancsmith/dotfiles/blob/af2b81d96aa4515c47a9a3131e610d3984bc5996/qutebrowser/config.py#L85-L90

**Comment by toofar** [comment]
> Is it possible to detect those URLs and automatically handle them in another application

I think you can do that with [the interceptor API](https://github.com/qutebrowser/qutebrowser/blob/main/qutebrowser/api/interceptor.py#L19): match the URL, block the request, exec external application. (There's a few examples of using the interceptor API around the place, here's [another one](https://github.com/qutebrowser/qutebrowser/issues/4793#issuecomment-1826421449)).

> rather only when I open them from another application

Hmm, not sure if you can detect that. Maybe you could compare the headers of a request made when invoked from an application vs a request from a website? Maybe the Origin header? Or maybe if it's an OAuth request you could have a look at the client_id query param?
